rosette/src/Export.cc

```diff
@@ -12,17 +12,161 @@
+    {"ExpandedLocation",    defaultObjectHandler,   ObjectCodePB::OT_ExpandedLocation},
+    {"Fixnum",              fixnumObjectHandler,    ObjectCodePB::OT_Fixnum},
+    {"Float",               floatObjectHandler,     ObjectCodePB::OT_Float},
+    {"FreeExpr",            defaultObjectHandler,   ObjectCodePB::OT_FreeExpr},
+    {"GotoExpr",            defaultObjectHandler,   ObjectCodePB::OT_GotoExpr},
+    {"IfExpr",              defaultObjectHandler,   ObjectCodePB::OT_IfExpr},
+    {"LabelExpr",           defaultObjectHandler,   ObjectCodePB::OT_LabelExpr},
+    {"LetExpr",             defaultObjectHandler,   ObjectCodePB::OT_LetExpr},
+    {"LetrecExpr",          defaultObjectHandler,   ObjectCodePB::OT_LetrecExpr},
+    {"MethodExpr",          defaultObjectHandler,   ObjectCodePB::OT_MethodExpr},
+    {"Proc",                defaultObjectHandler,   ObjectCodePB::OT_Proc},
+    {"ProcExpr",            defaultObjectHandler,   ObjectCodePB::OT_ProcExpr},
+    {"QuoteExpr",           defaultObjectHandler,   ObjectCodePB::OT_QuoteExpr},
+    {"RBLstring",           rblstringObjectHandler, ObjectCodePB::OT_RBLstring},
+    {"ReflectiveMethodExpr",defaultObjectHandler,   ObjectCodePB::OT_ReflectiveMethodExpr},
+    {"RequestExpr",         defaultObjectHandler,   ObjectCodePB::OT_RequestExpr},
+    {"SendExpr",            defaultObjectHandler,   ObjectCodePB::OT_SendExpr},
+    {"SeqExpr",             defaultObjectHandler,   ObjectCodePB::OT_SeqExpr},
+    {"SetExpr",             defaultObjectHandler,   ObjectCodePB::OT_SetExpr},
+    {"StdMthd",             defaultObjectHandler,   ObjectCodePB::OT_StdMthd},
+    {"Symbol",              symbolObjectHandler,    ObjectCodePB::OT_Symbol},
+    {"TblObject",           defaultObjectHandler,   ObjectCodePB::OT_TblObject},
+    {"Template",            defaultObjectHandler,   ObjectCodePB::OT_Template},
+    {"Tuple",               defaultObjectHandler,   ObjectCodePB::OT_Tuple},
+    {"TupleExpr",           defaultObjectHandler,   ObjectCodePB::OT_TupleExpr},
+};
+
+// Helper that searches the handler table for a specific object type
+ExportObjectHandling * findHandling(std::string type) {
+    int first = 0;
+    int last = sizeof(handlers)/sizeof(ExportObjectHandling) - 1;
+    int middle = (first+last)/2;
+
+    // Standard binary search of the list of handlers
+    while (first <= last) {
+        if (handlers[middle].obtypename == type) {
+            return &handlers[middle];
+        } else if (handlers[middle].obtypename < type) {
+            first = middle + 1;
+        } else {
+            last = middle - 1;
+        }
+
+        middle = (first + last)/2;
+    }
+    printf("Not found! %s isn't present in the list.\n", type.c_str());
+
+    return NULL;
+}
+
+std::set<uint64_t> codeblocks;  // temp debug
+
+// This is the protobuf that contains exported object code.
 ObjectCodePB::ObjectCode objectCode;
 
+// The collection routine that processes individual Code objects as they are compiled.
```

Comment:
*as they are compiled* - can you elaborate on that a bit?

Reply:
This just means that I call this function and collect the code for export when it is compiled.